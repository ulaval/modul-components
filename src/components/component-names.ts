export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
