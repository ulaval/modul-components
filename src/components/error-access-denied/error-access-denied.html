<m-error-template :skin="skin" :iconName="iconName" :title="title" :hints="hints" :links="links">
</m-error-template>
