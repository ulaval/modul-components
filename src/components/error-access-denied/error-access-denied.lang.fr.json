{
    "m-error-access-denied": {
        "title": "Vous n'avez pas les privilèges<br>pour voir cette page",
        "hint.primary": "Si vous croyez que cette situation est anormale, contacter le soutien technique.",
        "help": "Coordonnées du soutien technique"
    }
}
