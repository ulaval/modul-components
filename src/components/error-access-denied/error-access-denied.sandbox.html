<div>
    <h5>default</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-error-access-denied></m-error-access-denied>
    </div>

    <br />

    <h5>With custom title</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-error-access-denied :title="title"></m-error-access-denied>
    </div>

    <br />

    <h5>With many custom hints</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-error-access-denied :hints="manyHints"></m-error-access-denied>
    </div>

    <br />

    <h5>With many custom links</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-error-access-denied :links="manyLinks"></m-error-access-denied>
    </div>

    <br />

    <h5>Unexpected error popup</h5>
    <m-button v-m-popup:errorpopup>pop me an error</m-button>
    <m-modal ref="errorpopup" :negativeLink="false" ok-label="Fermer">
            <m-error-access-denied ></m-error-access-denied>
    </m-modal>
</div>
