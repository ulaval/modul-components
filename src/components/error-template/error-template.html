<article class="m-error-template" :class="{
    'm--is-skin-information': skin=='information',
    'm--is-skin-error': skin=='error',
    'm--is-skin-warning': skin=='warning'
    }">
    <h1 class="m-error-template__title" v-html="title"></h1>
    <div  class="m-error-template__icon-container" >
        <div class="m-error-template__icon" :style="styleObject" v-html="svg" :title="title" v-if="isSvg"></div>
        <m-icon class="m-error-template__icon" :name="iconName" :svg-title="title" :size="size + 'px'" v-else></m-icon>
    </div>
    <p ref="hint" class="m-error-template__hints" :class="{ 'm--no-content': !hasLinksAndSlot}" v-if="hasHints" v-for="hint in hints">{{hint}}</p>
    <div class="m-error-template__details" v-if="$slots['default']">
        <slot></slot>
    </div>
    <ul v-if="hasLinks" class="m-error-template__links">
        <li ref="link" v-for="link in links">
            <m-link class="m-error-template__link" icon-name="m-svg__chevron--right" mode="link" :url="link.url" :target="isTargetExternal(link.external)">{{link.label}}</m-link>
        </li>
    </ul>
</article>
