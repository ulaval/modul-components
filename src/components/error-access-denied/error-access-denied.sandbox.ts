import Vue, { PluginObject } from 'vue';
import { Component } from 'vue-property-decorator';

import { ERROR_ACCESS_DENIED_NAME } from '../component-names';
import { Link } from '../message-page/message-page';
import WithRender from './error-access-denied.sandbox.html';

@WithRender
@Component
export class MErrorAccessDeniedSandbox extends Vue {
    title: string = 'My custom access denied title';
    manyHints: string[] = ['My only custom hint', 'My second (long) custom hint.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas urna rhoncus ipsum congue lobortis. Mauris vel neque condimentum, dignissim lectus ac, vehicula justo. Aliquam nunc leo, tristique hendrerit aliquam sagittis, scelerisque quis libero. Integer non augue nec lacus aliquet porttitor in nec lacus. Cras ultrices tellus est, condimentum gravida orci pulvinar quis. Integer eget turpis arcu. Curabitur consequat porta urna, at hendrerit justo consectetur non. Aenean venenatis ornare nulla, a vulputate erat eleifend ut.'];
    manyLinks: Link[] = [new Link('The first custom link', 'http://www.ulaval.ca', true), new Link('The second custom link', 'http://www.google.com', true)];
}

const ErrorAccessDeniedSandboxPlugin: PluginObject<any> = {
    install(v, options): void {
        v.component(`${ ERROR_ACCESS_DENIED_NAME}-sandbox`, MErrorAccessDeniedSandbox);
    }
};

export default ErrorAccessDeniedSandboxPlugin;
