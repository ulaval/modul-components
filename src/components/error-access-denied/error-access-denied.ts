import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';

import { ModulVue } from '../../utils/vue/vue';
import { ERROR_ACCESS_DENIED_NAME } from '../component-names';
import ErrorTemplatePlugin, { Link, MErrorTemplateSkin } from '../error-template/error-template';
import I18nPlugin from '../i18n/i18n';
import LinkPlugin from '../link/link';
import WithRender from './error-access-denied.html';

@WithRender
@Component
export class MErrorAccessDenied extends ModulVue {

    @Prop({
        default: () => (Vue.prototype as any).$i18n.translate('m-error-access-denied:title')
    })
    public title: string;

    @Prop()
    public links: Link[];

    @Prop({
        default: () => [
            (Vue.prototype as any).$i18n.translate('m-error-access-denied:hint.primary')]
    })
    public hints: string[];

    readonly skin: string = MErrorTemplateSkin.Information;

    readonly svgName: string = 'm-svg__error-access-denied';
}

const ErrorAccessDeniedPlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug(ERROR_ACCESS_DENIED_NAME, 'plugin.install');
        v.use(I18nPlugin);
        v.use(LinkPlugin);
        v.use(ErrorTemplatePlugin);
        v.component(ERROR_ACCESS_DENIED_NAME, MErrorAccessDenied);
    }
};

export default ErrorAccessDeniedPlugin;
