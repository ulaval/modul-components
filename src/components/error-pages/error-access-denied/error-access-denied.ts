import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import { FRENCH, Messages } from '../../../utils/i18n/i18n';
import { ModulVue } from '../../../utils/vue/vue';
import { ERROR_ACCESS_DENIED_NAME } from '../../component-names';
import MessagePagePlugin, { Link } from '../../message-page/message-page';
import { MMessageState } from '../../message/message';
import WithRender from './error-access-denied.html';


@WithRender
@Component
export class MErrorAccessDenied extends ModulVue {

    @Prop({
        default: () => (Vue.prototype).$i18n.translate('m-error-access-denied:title')
    })
    public title: string;

    @Prop()
    public links: Link[];

    @Prop({
        default: () => [
            (Vue.prototype).$i18n.translate('m-error-access-denied:hint.primary')]
    })
    public hints: string[];

    readonly state: string = MMessageState.Information;

    readonly svgName: string = 'm-svg__error-access-denied';
}

const ErrorAccessDeniedPlugin: PluginObject<any> = {
    install(v, options): void {
        v.use(MessagePagePlugin);
        v.component(ERROR_ACCESS_DENIED_NAME, MErrorAccessDenied);
    }
};

export default ErrorAccessDeniedPlugin;
