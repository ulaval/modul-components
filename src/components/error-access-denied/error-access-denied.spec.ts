import { shallow, Wrapper } from '@vue/test-utils';

import { renderComponent } from '../../../tests/helpers/render';
import { ERROR_TEMPLATE_NAME } from '../component-names';
import { Link } from '../error-template/error-template';
import { MErrorAccessDenied } from './error-access-denied';

let wrapper: Wrapper<MErrorAccessDenied>;

const getStubs: any = () => {
    return {
        [ERROR_TEMPLATE_NAME]: '<div><slot /></div>'
    };
};

describe(`Page not found - test`, () => {
    describe(`Given default values`, () => {
        it(`Should render with default values`, async () => {
            wrapper = shallow(MErrorAccessDenied, { stubs: getStubs() });

            await expect(renderComponent(wrapper.vm)).resolves.toMatchSnapshot();
        });
    });
    describe(`Given custom values`, () => {
        it(`Should render with custom values provided`, async () => {
            const A_CUSTOM_TITLE: string = 'An error title.';
            const A_HINT: string = 'aHint';
            const A_LINK: Link = new Link('aLabel', 'anUrl');

            wrapper = shallow(MErrorAccessDenied, {
                stubs: getStubs(),
                propsData: {
                    title: A_CUSTOM_TITLE,
                    hints: [A_HINT],
                    links: [A_LINK]
                }
            });

            await expect(renderComponent(wrapper.vm)).resolves.toMatchSnapshot();
        });
    });
});
