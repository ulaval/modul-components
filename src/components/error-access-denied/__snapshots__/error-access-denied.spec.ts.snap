// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page not found - test Given custom values Should render with custom values provided 1`] = `<div state="information" svg-name="m-svg__error-access-denied" title="An error title." hints="aHint" links="[object Object]"></div>`;

exports[`Page not found - test Given default values Should render with default values 1`] = `<div state="information" svg-name="m-svg__error-access-denied" title="m-error-access-denied:title" hints="m-error-access-denied:hint.primary"></div>`;
